### PR TITLE
Testing environment for auth-testing

### PR DESCRIPTION
**Pull request by Dannikiskov**
Using the 'jest' framework, Stefan and I have created an testing environment for Typescript, for the sub folder 'auth-testing'.
The test file is found in services/auth-service/test with the name index.test.ts.
The testing approach is pretty self explanatory.
The result of the entire pipeline can be seen in the 'Actions' tab above.

### CODE REVIEW

File: services/auth-service/package.json [hunk 1]
```diff
@@ -2,16 +2,16 @@
...
-    "typescript": "^4.8.4"
+    "typescript": "^4.8.4",
+    "ts-jest": "^27.1.2"
   },
   "scripts": {
     "start": "node ./build/index.js",
     "build": "tsc",
     "dev": "nodemon",
-    "test": "tsc --noEmit"
+    "test": "jest"
```
Comment (viktorstrate) [inline services/auth-service/package.json:14]: Maybe change this to `tsc --noEmit && jest` such that type-checking is a part of the pipeline test?

Comment (Dannikiskov) [reply services/auth-service/package.json:14]: Sure! I didn't know there was a difference

Comment (Dannikiskov) [reply services/auth-service/package.json]: > Maybe change this to `tsc --noEmit && jest` such that type-checking is a part of the pipeline test?

Sure!

**Review by YousofMersal — COMMENTED** [review]
Ser fint ud fra mig af :+1:

File: services/auth-service/jest.config.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+module.exports = {
+    clearMocks: true,
+    moduleFileExtensions: ['js', 'ts'],
+    testMatch: ['**/*.test.ts'],
```
[the remark this answers is not in the document]
Comment (Setin2) [reply services/auth-service/jest.config.js:4]: I think so. Were we planning on using both typescript and javascript? Seems kinda weird to me.

File: .github/workflows/auth-testing.yaml [hunk 3]
```diff
@@ -0,0 +1,41 @@
+# The unit test file(s) should be placed inside a folder called test in the working directory specified
+name: auth-test
+on:
+  push:
+    branches: [ Stefan-Danni-auth-testing ]
+  pull_request:
+    branches: [ Stefan-Danni-auth-testing ]
```
Comment (jonasgroendahl) [inline .github/workflows/auth-testing.yaml:7]: husk at ændre det her inden merge

File: services/auth-service/tsconfig.json [hunk 4]
```diff
@@ -99,5 +99,6 @@
     /* Completeness */
     // "skipDefaultLibCheck": true,                      /* Skip type checking .d.ts files that are included with TypeScript. */
     "skipLibCheck": true                                 /* Skip type checking all .d.ts files. */
-  }
+  },
+     "exclude": ["node_modules", "**/*.test.ts"]
```
Comment (jonasgroendahl) [inline services/auth-service/tsconfig.json:103]: test shouldn't be typechecked by the `tsc`?

Comment (Setin2) [reply services/auth-service/tsconfig.json:103]: > the test shouldn't be typechecked by `tsc`? why

Think that's just how it was in the jest example we followed and we didn't bother changing it.

**Review by YousofMersal — APPROVED** [review]

**Pull request merged**